go.mod

```diff
@@ -1,10 +1,19 @@
 module knative.dev/net-gateway-api
 
-go 1.16
+go 1.17
+
+replace (
+	k8s.io/api v0.24.1 => k8s.io/api v0.23.8
+	k8s.io/apimachinery v0.24.1 => k8s.io/apimachinery v0.23.8
+	k8s.io/client-go v0.24.1 => k8s.io/client-go v0.23.8
+	k8s.io/code-generator v0.24.1 => k8s.io/code-generator v0.23.8
+	sigs.k8s.io/gateway-api v0.5.0-rc1 => sigs.k8s.io/gateway-api v0.4.0
```

Comment:
I think we should not close https://github.com/knative-sandbox/net-gateway-api/issues/306 yet (or track it by another issue) if we keep using v0.4.0.